packages/node_modules/@webex/webex-core/src/lib/services/services.js

```diff
@@ -702,50 +774,16 @@ const Services = WebexPlugin.extend({
 
     // Listen for configuration changes once.
     this.listenToOnce(this.webex, 'change:config', () => {
-      // Destructure the configuration object for the services plugin.
-      const {services} = this.webex.config;
-
-      // Validate that the services configuration exists.
-      if (services) {
-        // Check for discovery services.
-        if (services.discovery) {
-          // Format the discovery configuration into an injectable array.
-          const formattedDiscoveryServices = Object.keys(services.discovery)
-            .map((key) => ({
-              name: key,
-              defaultUrl: services.discovery[key]
-            }));
-
-          // Inject formatted services into services catalog.
-          catalog.updateServiceUrls('discovery', formattedDiscoveryServices);
-        }
-
-        // Check for allowed host domains.
-        if (services.allowedDomains) {
-          // Store the allowed domains as a property of the catalog.
-          catalog.setAllowedDomains(services.allowedDomains);
-        }
-
-        // Set `validateDomains` property to match configuration
-        this.validateDomains = services.validateDomains;
-      }
+      this.initConfig();
     });
 
     // wait for webex instance to be ready before attempting
     // to update the service catalogs
     this.listenToOnce(this.webex, 'ready', () => {
-      /* eslint-disable camelcase */
-      if (this.webex.credentials.canAuthorize) {
-        this.updateServices()
-        // this catch prevents crashing in unique situations found
-        // primarily in unit testing with the karma suite.
-          .catch(
-            () => this.logger.warn('services: catalog retrieval failed w/auth')
-          );
-      }
-      else {
-        this.collectPreauthCatalog({email: this.webex.config.email});
-      }
+      this.initServiceCatalogs()
+        .catch((error) => this.logger.info(
+          `services: failed to init initial services, ${error.message}`
```

Comment:
One thing I'd like to think about especially for v2 is a way to communicate errors with the developer outside of just console logging.  Something like `this.webex.emit('error')`.

Reply:
I agree. Along with the larger list of things v2 will have, this is among the highest priority to me.